### PR TITLE
docs: 📚 rewrite docs

### PR DESCRIPTION
**Pull request by mariohamann**
## Description:
Complete rewrite of our documentation in Storybook. For more information pls check on https://solid-design-system.fe.union-investment.de/x.x.x/storybook/index.html?path=/docs/docs-general-version-4-announcement--docs

## Definition of Reviewable:

- [ ] Documentation is created/updated
- [ ] Migration Guide is created/updated
- [ ] E2E tests (features, a11y, bug fixes) are created/updated
- [ ] Stories (features, a11y) are created/updated
- [ ] relevant tickets are linked

### CODE REVIEW

File: packages/components/src/components/teaser/teaser.stories.ts [hunk 1]
```diff
@@ -35,711 +43,149 @@ export default {
...
+        Quis ut ex cupidatat proident cillum ullamco ea aute ad laborum aliqua incididunt sint ipsum. Elit enim
+        reprehenderit aliquip officia in minim. Eu ipsum pariatur dolor. Do ex in cupidatat anim aliqua sint voluptate
+        sunt nulla incididunt. Cupidatat officia reprehenderit est cupidatat et id officia ut. Exercitation id pariatur
+        elit occaecat ad Lorem nisi sunt pariatur do aute aliqua magna irure incididunt.
+      </p>
+    </sd-teaser>
+  `
 };
 
 /**
```
Comment (coraliefeil) [inline packages/components/src/components/teaser/teaser.stories.ts:69]: Pls move variant section further up as the 1st section to appear underneath the props.

File: packages/components/src/components/accordion/accordion.stories.ts [hunk 2]
```diff
@@ -2,20 +2,31 @@
...
+ *
+ *  **Related templates**:
+ * - [Accordion Group](?path=/docs/templates-accordion-group--docs)
+ *
+ *
+ *  **Related components**:
+ * - [sd-accordion-group](?path=/docs/components-sd-accordion-group--docs)
+ *
+ */
+
```
Comment (coraliefeil) [inline packages/components/src/components/accordion/accordion.stories.ts:22]: - Use the `summary` slot to add text. 
- Use the `default` slot to add content.

File: packages/components/src/components/divider/divider.stories.ts [hunk 3]
```diff
@@ -1,149 +1,67 @@
...
- * This shows sd-divider in its default state.
- */
 
 export const Default = {
-  render: (args: any) => {
-    return generateTemplate({ args });
-  }
+  name: 'Default',
+  render: () => html`<sd-divider orientation="horizontal" class="w-[120px]"></sd-divider>`
 };
```
Comment (mariohamann) [inline packages/components/src/components/divider/divider.stories.ts:24]: @auroraVasconcelos Please make dynamic again. Do it directly here in the branch. Please comment, when done.

File: packages/components/src/templates/button-with-badge.stories.ts [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+  parameters: {
+    chromatic: { disableSnapshot: true },
+    design: {
+      type: 'figma',
+      url: 'https://www.figma.com/file/hER2N8wZXhiTrdlDXyrdUt/Quote?type=design&node-id=1001-4293&mode=design&t=Xywhix1rQMoatokH-0'
+    }
+  }
+};
+/**
+ * Here are some examples of sd-button working with sd-badge.
```
Comment (coraliefeil) [inline packages/components/src/templates/button-with-badge.stories.ts:16]: Examples of sd-button working with sd-badge:

File: packages/components/src/templates/button-with-badge.stories.ts [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+ */
+export const Default = {
+  render: () => html`
+    <section class="flex p-4 gap-10 items-center">
```
Comment (coraliefeil) [inline packages/components/src/templates/button-with-badge.stories.ts:20]: Pls see [figma template](https://www.figma.com/design/YDktJcseQIIQbsuCpoKS4V/Component-Docs?node-id=2501-4280&t=8XG4ATjdx0TEu0Dm-11)

And adjust what we show accordingly.

Inverted can be just added directly underneath. No extra headline/subline neccessary.

File: packages/components/src/components/accordion/accordion.stories.ts [hunk 6]
```diff
@@ -2,20 +2,31 @@
...
-import { waitUntil } from '@open-wc/testing-helpers';
-import cx from 'classix';
 
 const { argTypes, parameters } = storybookDefaults('sd-accordion');
 const { overrideArgs } = storybookHelpers('sd-accordion');
 const { generateTemplate } = storybookTemplate('sd-accordion');
 
+/**
+ *
+ * Shows a brief summary and expands to display additional content.
```
Comment (MartaPintoTeixeira) [inline packages/components/src/components/accordion/accordion.stories.ts:12]:
```suggestion
 * Used to show a brief summary and expands to display additional content.
```

File: packages/components/src/components/brandshape/brandshape.stories.ts [hunk 7]
```diff
@@ -1,15 +1,25 @@
...
 import { storybookDefaults, storybookHelpers, storybookTemplate } from '../../../scripts/storybook/helper';
-import type { ConstantDefinition } from '../../../scripts/storybook/helper';
 
 const { argTypes, parameters } = storybookDefaults('sd-brandshape');
 const { overrideArgs } = storybookHelpers('sd-brandshape');
 const { generateTemplate } = storybookTemplate('sd-brandshape');
 
+/**
+ *
+ * Creates a container with brand character and visually emphasizes the content.
```
Comment (MartaPintoTeixeira) [inline packages/components/src/components/brandshape/brandshape.stories.ts:13]:
```suggestion
 * Used to create a container with brand character and visually emphasizes the content.
```

File: packages/components/src/components/carousel/carousel.stories.ts [hunk 8]
```diff
@@ -2,14 +2,25 @@
...
 import { html } from 'lit';
 import { storybookDefaults, storybookHelpers, storybookTemplate } from '../../../scripts/storybook/helper';
-import { waitUntil } from '@open-wc/testing-helpers';
 
 const { argTypes, parameters } = storybookDefaults('sd-carousel');
 const { overrideArgs } = storybookHelpers('sd-carousel');
 const { generateTemplate } = storybookTemplate('sd-carousel');
 
+/**
+ * Displays an arbitrary number of content slides along a horizontal axis.
```
Comment (MartaPintoTeixeira) [inline packages/components/src/components/carousel/carousel.stories.ts:11]:
```suggestion
 * Used to display an arbitrary number of content slides along a horizontal axis.
```

File: packages/components/src/components/expandable/expandable.test.stories.ts [hunk 9]
```diff
@@ -0,0 +1,267 @@
...
+  component: 'sd-expandable',
+  args: overrideArgs([
+    { type: 'slot', name: 'default', value: '<div class="slot slot--border slot--text h-16">Default slot</div>' }
+  ]),
+  argTypes,
+  parameters: { ...parameters }
+};
+
+/**
+ * Expandable shows a brief summary and expands to show additional content.
```
Comment (MartaPintoTeixeira) [inline packages/components/src/components/expandable/expandable.test.stories.ts:30]:
```suggestion
 * Used to help reducing visible content by concealing part of it, with an option for users to reveal more as needed.
```

File: packages/components/src/components/spinner/spinner.stories.ts [hunk 10]
```diff
@@ -1,75 +1,66 @@
 import '../../solid-components';
 import { html } from 'lit-html';
 import { storybookDefaults, storybookTemplate } from '../../../scripts/storybook/helper';
-import { withActions } from '@storybook/addon-actions/decorator';
 
 const { argTypes, args, parameters } = storybookDefaults('sd-spinner');
 const { generateTemplate } = storybookTemplate('sd-spinner');
 
+/**
+ * Visual indicator with a looping animation that shows loading is in process.
```
Comment (MartaPintoTeixeira) [inline packages/components/src/components/spinner/spinner.stories.ts:9]:
```suggestion
 * Used as a visual indicator with a looping animation that shows loading is in process.
```

File: packages/components/src/styles/chip/chip.stories.ts [hunk 11]
```diff
@@ -1,36 +1,23 @@
...
 import { storybookDefaults, storybookHelpers, storybookTemplate } from '../../../scripts/storybook/helper';
 
-const { argTypes, parameters } = storybookDefaults('sd-chip');
+const { argTypes } = storybookDefaults('sd-chip');
 const { overrideArgs } = storybookHelpers('sd-chip');
 const { generateTemplate } = storybookTemplate('sd-chip');
 
 /**
- * A small, non-interactive label the represents a status, property or meta-data.
+ * A small, non-interactive label that represents a status, property or meta-data.
```
Comment (MartaPintoTeixeira) [inline packages/components/src/styles/chip/chip.stories.ts:10]:
```suggestion
 * Used as a small, non-interactive label that represents a status, property or meta-data.
```

File: packages/components/src/styles/flag/flag.stories.ts [hunk 12]
```diff
@@ -1,36 +1,30 @@
...
-
 import { storybookDefaults, storybookHelpers, storybookTemplate } from '../../../scripts/storybook/helper';
 
 const { argTypes, parameters } = storybookDefaults('sd-flag');
 const { overrideArgs } = storybookHelpers('sd-flag');
 const { generateTemplate } = storybookTemplate('sd-flag');
 
 /**
- * A small, non-interactive label the represents a category.
+ * A small, non-interactive label that represents a category.
```
Comment (MartaPintoTeixeira) [inline packages/components/src/styles/flag/flag.stories.ts:10]:
```suggestion
 * Used as a small, non-interactive label that represents a category.
```

File: packages/components/src/components/teaser/teaser.stories.ts [hunk 13]
```diff
@@ -6,7 +6,15 @@ const { argTypes, parameters } = storybookDefaults('sd-teaser');
 const { overrideArgs } = storybookHelpers('sd-teaser');
 const { generateTemplate } = storybookTemplate('sd-teaser');
 
+/**
+ * Teasers can be used to group related subjects in a container.
```
Comment (MartaPintoTeixeira) [inline packages/components/src/components/teaser/teaser.stories.ts:10]:
```suggestion
 * Used to group related subjects in a container.
```

File: packages/components/src/components/drawer/drawer.stories.ts [hunk 14]
```diff
@@ -1,23 +1,26 @@
...
-import { userEvent } from '@storybook/test';
-import { waitUntil } from '@open-wc/testing-helpers';
-import cx from 'classix';
-
 const { argTypes, parameters } = storybookDefaults('sd-drawer');
 const { generateTemplate } = storybookTemplate('sd-drawer');
 const { overrideArgs } = storybookHelpers('sd-drawer');
 
+/**
+ * A panel that slides out from the side of the screen which contains a set of information or actions.
```
Comment (MartaPintoTeixeira) [inline packages/components/src/components/drawer/drawer.stories.ts:9]:
```suggestion
 * Used as a panel that slides out from the side of the screen which contains a set of information or actions.
```

File: packages/components/src/components/button/button.stories.ts [hunk 15]
```diff
@@ -2,14 +2,21 @@
...
 import { storybookDefaults, storybookHelpers, storybookTemplate } from '../../../scripts/storybook/helper';
-import { userEvent } from '@storybook/test';
-import { waitUntil } from '@open-wc/testing-helpers';
 import { withActions } from '@storybook/addon-actions/decorator';
 const { argTypes, parameters } = storybookDefaults('sd-button');
 const { overrideArgs } = storybookHelpers('sd-button');
 const { generateTemplate } = storybookTemplate('sd-button'); // Replace with your custom element tag
 
+/**
+ * **Allows users to perform actions with a single click.**
```
Comment (MartaPintoTeixeira) [inline packages/components/src/components/button/button.stories.ts:11]:
```suggestion
 * **Used to allow users to perform actions with a single click. 
```

File: packages/components/src/components/link/link.stories.ts [hunk 16]
```diff
@@ -2,19 +2,26 @@
...
-import { waitUntil } from '@open-wc/testing-helpers';
 import { withActions } from '@storybook/addon-actions/decorator';
 
 const { argTypes } = storybookDefaults('sd-link');
 const { generateTemplate } = storybookTemplate('sd-link');
 const { overrideArgs } = storybookHelpers('sd-link');
 
+/**
+ *
+ * Allows users to navigate to another location or perform actions like e.g. downloading files.
```
Comment (MartaPintoTeixeira) [inline packages/components/src/components/link/link.stories.ts:13]:
```suggestion
 * Used to allow users to navigate to another location or perform actions like e.g. downloading files. 
```

File: packages/components/src/styles/copyright/copyright.stories.ts [hunk 17]
```diff
@@ -1,16 +1,16 @@
...
-
 import { storybookDefaults, storybookHelpers, storybookTemplate } from '../../../scripts/storybook/helper';
 
 const { argTypes, parameters } = storybookDefaults('sd-copyright');
 const { overrideArgs } = storybookHelpers('sd-copyright');
 const { generateTemplate } = storybookTemplate('sd-copyright');
 
 /**
- * Add a copyright at the bottom of a image.
+ * Use the CSS property `--copyright` and the class `sd-copyright` in the parent of any element to set a copyright text.
```
Comment (MartaPintoTeixeira) [inline packages/components/src/styles/copyright/copyright.stories.ts:9]:
```suggestion
 * Used to be displayed at the bottom of an image for example.
```

File: packages/components/src/components/textarea/textarea.stories.ts [hunk 18]
```diff
@@ -30,400 +32,286 @@ export default {
...
+      <sd-textarea size="lg" label="Label" rows="8" spellcheck></sd-textarea>
+
+      <sd-textarea size="lg" label="Label" rows="12" spellcheck></sd-textarea>
+    </div>
+  `
 };
 
 /**
- * Use the readonly attribute to render a textarea as readonly.  Interaction is enabled, but the textarea cannot be edited.  Events will be fired.
+ * Use the`label`attribute to give the input an accessible label.
```
Comment (coraliefeil) [inline packages/components/src/components/textarea/textarea.stories.ts:77]: Use the "label" attribute to provide an accessible label for the textarea element.
Use the "label" slot to add custom markup.

Comment (smfonseca) [reply packages/components/src/components/textarea/textarea.stories.ts:77]: Addressed at [#1367](https://github.com/solid-design-system/solid/pull/1367)

File: packages/components/src/components/carousel/carousel.ts [hunk 19]
```diff
@@ -318,6 +347,8 @@ export default class SdCarousel extends SolidElement {
         slide.style.setProperty('scroll-snap-align', 'none');
       }
     });
+
+    // this.handleScrollEnd();
```
Comment (karlbaumhauer) [inline packages/components/src/components/carousel/carousel.ts:351]: looks like a left-over

**Review by smfonseca — APPROVED** [review]

**Pull request merged**